src/incidencehandler.cpp

```diff
@@ -193,17 +193,30 @@ void IncidenceHandler::copyIncidenceProperties(KCalCore::Incidence::Ptr dest, co
     KDateTime origCreated = dest->created();
     KDateTime origLastModified = dest->lastModified();
 
-    // Recurrences need to be copied through serialization, else they are not recreated.
-    // Do this first to avoid overriding other copied properties.
+    // Copy recurrence information if required.
     if (*(dest->recurrence()) != *(src->recurrence())) {
-        dest->clearRecurrence();
-        QBuffer buffer;
-        buffer.open(QBuffer::ReadWrite);
-        QDataStream out(&buffer);
-        out << *src.data();
-        QDataStream in(&buffer);
-        buffer.seek(0);
-        in >> *dest.data();
+        dest->recurrence()->clear();
+
+        KCalCore::Recurrence *dr = dest->recurrence();
+        KCalCore::Recurrence *sr = src->recurrence();
+
+        // recurrence rules and dates
+        KCalCore::RecurrenceRule::List srRRules = sr->rRules();
+        for (QList<KCalCore::RecurrenceRule*>::const_iterator it = srRRules.constBegin(), end = srRRules.constEnd(); it != end; ++it) {
+            KCalCore::RecurrenceRule *r = new KCalCore::RecurrenceRule(*(*it));
```

Comment:
kcal documentation is atrocious, but addRRule takes ownership, so if we don't use operator new here (but instead just do:
    dr->addRRule(*it);
we eventually get a crash.